homeassistant/components/intelliclima/entity.py

```diff
@@ -0,0 +1,71 @@
+"""Platform for shared base classes for sensors."""
+
+from pyintelliclima import IntelliClimaC800, IntelliClimaECO
+
+from homeassistant.helpers.device_registry import (
+    CONNECTION_BLUETOOTH,
+    CONNECTION_NETWORK_MAC,
+    DeviceInfo,
+)
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import IntelliClimaCoordinator
+
+
+class IntelliClimaEntity(CoordinatorEntity[IntelliClimaCoordinator]):
+    """Define a generic class for IntelliClima entities."""
+
+    _attr_attribution = "Data provided by unpublished IntelliClima API"
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: IntelliClimaCoordinator,
+        device: IntelliClimaECO | IntelliClimaC800,
+        description: EntityDescription,
+    ) -> None:
+        """Class initializer."""
+        super().__init__(coordinator=coordinator)
+
+        self.entity_description = description
+        self._attr_unique_id = f"{device.id}_{description.key}"
+
+        # Make this HA "device" use the IntelliClima device name.
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.id)},
+            manufacturer="Fantini Cosmi",
+            name=device.name,
+            serial_number=device.crono_sn,
+        )
+
+        self._device_id = device.id
+        self._device_sn = device.crono_sn
+
+
+class IntelliClimaECOEntity(IntelliClimaEntity):
```

Comment:
If we support a single device, we should only have one class. We don't need to create a class hierarchy yet. This simplifies reviews. As we add more devices, we can consider how to make our code more generic.